pkg/kn/flags/podspec_helper.go

```diff
@@ -97,6 +97,35 @@ func UpdateEnvVars(spec *corev1.PodSpec,
 	return nil
 }
 
+// UpdateEnvsFromFile gives the configuration all the env var values listed in the given file.
+// It is similar to UpdateEnvVars with the exception of reading the environment variables from
+// a file and updating it in the container environment variables.
+func UpdateEnvsFromFile(spec *corev1.PodSpec, filepath string) error {
+    envsFromFile, rmEnvFromFile, err := util.GetEnvsFromFile(filepath, "=")
+    if err != nil {
+        return err
+    }
+
+	container := containerOfPodSpec(spec)
+	allEnvsToUpdate := util.NewOrderedMap()
+	envIterator := envsFromFile.Iterator()
+
+	envKey, envValue, done := envIterator.NextString()
+	for done {
+        allEnvsToUpdate.Set(envKey, corev1.EnvVar{
+            Name:  envKey,
+            Value: envValue,
+        })
+        envKey, envValue, done = envIterator.NextString()
+    }
+
+    updated := updateEnvVarsFromMap(container.Env, allEnvsToUpdate)
+    updated = removeEnvVars(updated, rmEnvFromFile)
+	container.Env = updated
+
```

Comment:
is there a way you can use this func `UpdateEnvVars` instead of duplicating the same logic here ?

Reply:
Thanks @itsmurugappan for your feedback and sorry for the late response. I think I solved this but maybe you can find a more elegant way. 
You can see the updates here:  https://github.com/knative/client/blob/fa9ef7f886ff6cefcc6866b151e1f068ce582157/pkg/kn/flags/podspec.go#L177-L192